hashintel/hash
# Use Advanced Visibility in Postgres for Temporal

## 🌟 What is the purpose of this PR?



This PR enables Advanced Visibility for our local Temporal setup in anticipation for using the a similar setup in production.
From Temporal version 1.20+ it is possible to use Postgres 12+ as an Advanced Visibility store, which means we don't need to run Cassandra/OpenSearch for Temporal.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1204274556967336/1204521571620726/f) _(internal)_

## 🚫 Blocked by




## 🔍 What does this change?




- Changes the setup/migrate scripts slightly to switch to pg12+
- Use `TEMPORAL_ADDRESS` instead of `TEMPORAL_CLI_ADDRESS` as the latter is deprecated.
- Parameterize the Temporal docker container version to allow us to easily update the version we're running

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:













### 📜 Does this require a change to the docs?




The changes in this PR:






### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:







## ⚠️ Known issues



## 🐾 Next steps


Prod infra for temporal [Asana task](https://app.asana.com/0/1204274556967336/1204521571620726/f) (_internal_)

## 🛡 What tests cover this?



Not directly tested, but local dev/ci would fail if the compose is wrong.

## ❓ How to test this?



1.  Checkout the branch
1.  Start external services with an OpenAI key configured
1.  Run `temporal workflow execute -t ai -w foo --type DemoWorkflow -i '"How many eggs can you fit in a basket"'` to submit a workflow
1.  Verify that the advanced visibility store is working by running `temporal workflow list --query "TaskQueue = 'ai' AND ExecutionStatus = 2"`. This command fails if advanced visibility is not configured for the operator

## 📹 Demo

![image](https://github.com/hashintel/hash/assets/6988668/8d59be51-25e2-4418-ab25-7d093fc9417b)
